### ISSUE TITLE
Cannot generate a pickler for a class named 'Unit' when not namespaced.

### ISSUE THREAD

**Issue by eaceaser** [scala/pickling#42]
## steps

To reproduce, run from the REPL:

```scala
scala> import scala.pickling._, Defaults._
import scala.pickling._
import Defaults._

scala> import json._
import json._

scala> object Container { case class Unit(x: Int) }
defined module Container

scala> import Container._
import Container._

scala> case class C(x: Unit)
defined class C

scala> val c = C(Unit(3))
c: C = C(Unit(3))

scala> c.pickle
```

## problem (0.10.0)

```scala
scala> c.pickle
<console>:24: error: Cannot generate a pickler for C. Recompile with -Xlog-implicits for details
              c.pickle
                ^
```

## problem from the original report

fails with:

```scala
<console>:20: pickling.this.SPickler.genPickler is not a valid implicit value for scala.pickling.SPickler[C.type] because:
hasMatchingSymbol reported error: type mismatch;
 found   : scala.Unit
 required: Container.Unit
              C.pickle
```

## expectations

Note that redefining C with a fully qualified path, without importing Container._:

```
case class C(x: Container.Unit)
val c = C(Container.Unit(3))
c.pickle
```

evaluates to 

```
JSONPickle({
  "tpe": "C",
  "x": {
    "x": 3
  }
})
```

which is correct.

**Comment by eed3si9n** [comment]
I can reproduce the problem using Pickling 0.10.0.

``` scala
scala-pickling> set scalacOptions in (Compile, console) += "-Xlog-implicits"
[info] Defining scala-pickling/compile:console::scalacOptions
[info] The new value will be used by scala-pickling/compile:console, scala-pickling/compile:consoleQuick
[info] Reapplying settings...
[info] Set current project to scala-pickling (in build file:/pickling/)
scala-pickling> console
[info] Starting scala interpreter...
[info] 
Welcome to Scala version 2.10.4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.7.0_51).
Type in expressions to have them evaluated.
Type :help for more information.

scala> import scala.pickling._, Defaults._
import scala.pickling._
import Defaults._

scala> import json._
import json._

scala> object Container { case class Unit(x: Int) }
defined module Container

scala> import Container._
import Container._

scala> case class C(x: Unit)
defined class C

scala> val c = C(Unit(3))
c: C = C(Unit(3))

scala> c.pickle
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[C] because:
stepping aside: repeating itself
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[C] because:
stepping aside: repeating itself
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Container.Unit] because:
stepping aside: repeating itself
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Container.Unit] because:
stepping aside: repeating itself
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Int] because:
stepping aside: repeating itself
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Int] because:
stepping aside: there are other candidates
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Container.Unit] because:
hasMatchingSymbol reported error: type mismatch;
 found   : scala.Unit
 required: Container.Unit
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Container.Unit] because:
stepping aside: repeating itself
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Container.Unit] because:
stepping aside: repeating itself
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Container.Unit] because:
hasMatchingSymbol reported error: type mismatch;
 found   : scala.Unit
 required: Container.Unit
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Container.Unit] because:
stepping aside: repeating itself
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Container.Unit] because:
stepping aside: repeating itself
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[Container.Unit] because:
hasMatchingSymbol reported error: type mismatch;
 found   : scala.Unit
 required: Container.Unit
              c.pickle
                ^
<console>:24: genPickler is not a valid implicit value for scala.pickling.Pickler[C] because:
hasMatchingSymbol reported error: Cannot generate a pickler for Container.Unit. Recompile with -Xlog-implicits for details
              c.pickle
                ^
<console>:24: error: Cannot generate a pickler for C. Recompile with -Xlog-implicits for details
              c.pickle
                ^
```

**Comment by jsuereth** [comment]
We recently cleaned up a lot of macro hygiene issues.  This is now fixes in the 0.11.x branch.
